src/lib/y2storage/blk_device.rb

```diff
@@ -395,7 +426,15 @@ def component_of
     # @return [Array<String>]
     def component_of_names
       # So far, all the possible elements on the array respond to #name
-      component_of.map(&:name)
+      component_of.map do |dev|
+        if dev.respond_to?(:name)
+          dev.name
+        elsif dev.respond_to?(:uuid)
```

Comment:
I don't know if displaying an UUID will help the users to know that they are removing the cset from a given bcache device. Something including the bcache name would probably be more informative.

On the other hand, raising an exception here may be a little bit radical. :wink:

Reply:
Problem is that bcache caching device can be used in multiple bcaches and also for flash only volume, so the name can be really big. So I am not sure what would the best name for users.

Reply:
So what about something like this?

 * "cache set -bcache0-"
 * "cache set -bcache0, bcache3-"
 * "cache set" (*for flash only devices*)

That's still more informative about the devices to be affected than something like "0f517398-11ba-4c6c-9a5d-52b76c488303".